com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs

```diff
@@ -22,14 +23,195 @@ public LightweightRenderPipeline(LightweightRenderPipelineAsset asset)
 
 namespace UnityEngine.Rendering.Universal
 {
+    // A collection of profiling samplers, cache, helpers and convenience functions for Universal profiling.
+    // Keeps most of the profiling related things organized in one place for editability.
+    // Static global state, mostly read/sometimes write, non-thread safe (like most things in Universal).
+    //
+    // Example:
+    // // Cached CPU
+    // using var profScope = UniversalProfilingCache.GetCPUScope("example name");
+    // // Static GPU
+    // using( UniversalProfilingCache.GetGPUScope( cmd, UniversalProfilingScope.Pipeline.ExampleStaticSampler ) ) {}
+    internal static class UniversalProfilingCache
+    {
+        private static Dictionary<string, ProfilingSampler> m_StringSamplerCache = new Dictionary<string, ProfilingSampler>();
+        private static Dictionary<int, ProfilingSampler>    m_IntSamplerCache    = new Dictionary<int, ProfilingSampler>();
+        private static Dictionary<int, ProfilingSampler>    m_CameraSamplerCache = new Dictionary<int, ProfilingSampler>();
+
+        // A static (fast) fallback profiling name.
+        public static readonly ProfilingSampler m_UnknownSampler = new ProfilingSampler("Unknown");
+
+        static UniversalProfilingCache()
+        {
+            // TODO: with Net 5.0 support
+            // m_SamplerCache.EnsureCapacity(64);
+        }
+
+        // Creates a ProfilingScope for immediate CPU profiling only.
+        // "Inl_" CPU time outside of any command buffers.
+        // TODO: should we add a verbosity level? something like GetCPUScope(string name, ProfilingVerbosity verbosity = default)
+        public static ProfilingScope GetCPUScope(string scopeName)
+        {
+            return new ProfilingScope(null, TryGetOrAddSampler(scopeName));
+        }
+
+        // Creates a ProfilingScope for GPU and CPU profiling into a command buffer.
+        // Which show the GPU and the CPU time for executing the command buffer.
+        // Also creates "Inl_" CPU time scope outside of the command buffer.
+        public static ProfilingScope GetGPUScope(CommandBuffer cmd, string scopeName)
```

Comment:
Because it's nice API :) but I'm a bit worried too. No need to edit in multiple places, much faster to add scopes or do temporary ones if you are testing something. Also you can generate text, not enums. I suppose you can create a ProfilingSampler with a custom text too. That involves some boilerplate though.
Cached and hashed string of ~20 chars, took 0.002ms, while allocating a new ProfilingSampler took 0.01ms. So hashed/cached string is still 5x faster (on my machine).
So 100 scopes would add 0.2ms of overhead.